### PR TITLE
Adding documentation (plus a few TEB parameter tweaks)

### PR DESCRIPTION
**Pull request by kob51**
Went thru and added more formalized readme's to every package I could. Lmk if more info is needed anywhere...

I was a little confused by what was written in the `autovalet_description` package's readme, so someone please take a look at that and lmk if it needs updating.

Also renamed a few packages, which is why so many files have been touched:
`valet_husky` --> `autovalet_husky`
`lane_detection` --> `autovalet_lane_detection`

Because of the package renames, you will need to do a `catkin clean` and then a fresh `catkin build` so that you are no longer looking for packages that won't exist anymore.

### CODE REVIEW

File: autovalet_description/readme.MD [hunk 1]
```diff
@@ -1,3 +1,5 @@
+# autovalet_description
+
 This is the autovalet husky URDF description package. We have a modified version of the husky with a LiDAR and 2 d435s on it. The huskyValet_urdf launch file calls xacro parser onto the modified urdf passed to it. Also, it runs robot_state publisher and joint_state publisher for standalone vizualization purposes.
```
Comment (SubramanianKrish) [inline autovalet_description/readme.MD:4]: I think the readme here should go sth like this:
This is the autovalet husky URDF package. We have a modified version of the husky with a LiDAR and 2 Realsense D435s on it. We call the standard husky URDF with out autovalet.urdf.xacro as an extra. Our sensor transforms are defined in the xacro file. The URDF is published to ROS by launching the autovalet_description.launch. In order to view the robot model generated by the URDF, please launch robot state publisher and the joint state publisher along with the autovalet_description.launch file.

**Review by SubramanianKrish — DISMISSED** [review]
Check the autovalet_description readme. Feel free to edit it out

File: autovalet_lane_detection/README.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# autovalet_lane_detection
+
+This package contains our lane detection pipeline. The output of the lane detection will be used projected on the costmap, so that valid goals lying within the lane can be sent to the planner.
```
Comment (evamo0508) [inline autovalet_lane_detection/README.md]: think we should add one line: `roslaunch autovalet_lane_detection lane_detector.launch`

Comment (kob51) [reply autovalet_lane_detection/README.md]: Adding now

**Review by SubramanianKrish — DISMISSED** [review]

**Review by evamo0508 — APPROVED** [review]

**Review by SubramanianKrish — APPROVED** [review]

**Pull request merged**